services/src/main/java/org/keycloak/protocol/saml/profile/ecp/authenticator/HttpBasicAuthenticator.java

```diff
@@ -39,8 +39,12 @@ public void authenticate(final AuthenticationFlowContext context) {
                 final boolean valid = context.getSession().userCredentialManager().isValid(realm, user, UserCredentialModel.password(password));
 
                 if (valid) {
-                    context.getClientSession().setAuthenticatedUser(user);
-                    context.success();
+                    if (user.isEnabled()) {
+                        context.getClientSession().setAuthenticatedUser(user);
+                        context.success();
+                    } else {
+                        userDisabledFailure(context, realm, user);
+                    }
```

Comment:
OK - so I updated the userDisabledFailure to do what it was doing before in the HttpBasicAuthenticator: letting users in.  Otherwise, the refactoring in place is just to expose various action components to the Docker child class.  Will that fly?